src/odoc/page-index.mld

```diff
@@ -0,0 +1,252 @@
+
++ {{:#overview} What is [odoc] really? }
++ {{:#flow} How does it work? }
++ {{:#deep-dive} Digging Deeper }
+
+{2:overview What is [odoc] really?}
+
+[odoc] is not your regular documentation tool.
+
+Instead of source code, it works with {i compilation units}; that is, it works
+on compiler outputs and turns them into compiled documentation, which then
+becomes gorgeous HTML.
+
+A regular [odoc] execution transforms [cmt], [cmti], and [mld] files into [odoc]
+files, and then turns those into [html] files. Roughly like this:
+
+{[
+open List
+
+[
+  Game.cmti;
+  Player.cmti;
+  Monster.cmti;
+  page-intro.mld;
+]
+|> map ~f:compile_to_odoc  (* here we have Game.odoc, Player.odoc, etc *)
+|> iter ~f:compile_to_html (* here we have Game/index.html, and so on  *)
+
+]}
+
+This means that [page-intro.mld] will be compiled to [page-intro.odoc] which
+in turn will become [intro.html].
+
+Similarly, [Game.cmti] will be compiled to [Game.odoc] which in turn will become
+[Game/index.html].
+
+{2:flow How does it work?}
+
+[odoc] is built in a very modular fashion, with the code is split into several
+modules that take care of mostly orthogonal concerns.
+
+Instead of namedropping them, we will describe a use-case and how they connect to
+each other as we analyze its execution superficially.
+
+We will begin the flow with an [odoc] command that compiles a single [cmti] file,
+[Player.cmti], into its corresponding [html] file. Then we will compile the intro
+documentation file into html.
+
++ {{:#step-1} Compiling Player.cmti to Player.odoc }
++ {{:#step-2} Compiling Player.odoc to Player/index.html }
++ {{:#step-3} Compiling page-intro.mld to page-intro.odoc }
++ {{:#step-4} Compiling page-intro.odoc to intro.html }
+
+Off we go!
+
+
+{4:step-1 1 — Compiling Player.cmti to Player.odoc}
+
+Superficially, what we need to do is straighforward.
+
+{[
+# We must tell odoc what the name of this Package is!
+λ odoc compile --package Game -o ./src/Player.odoc ./src/Player.cmti
+λ ls src
+Player.cmti Player.mli Player.odoc
+]}
+
+Voila! We get a [Player.odoc] right where we expected it. But what really just
+happened?
+
++ In {!Main} the command was parsed and a decision was made to {i compile} this
+  file into an [odoc] file
++ Compilation is orchestrated by {!Main.Compile.compile}, that based on the
+  input extension will delegate to one of many {!Compile} functions. In this
+  case {!Compile.cmti}
++ The [cmti] file is read by {!Loader.read_cmti} into a {!Model.Root.t} and a
+  {!Compilation_unit.t} is created
++ This compilation unit is then turned into an {!Env.t} (environment),
+  expanding all found references between modules
++ And lastly {!Compilation_unit.save} takes care of saving this compilation
+  unit into the [Player.odoc] file in marshalled format.
+
+
+{4:step-2 2 — Compiling Player.odoc to Player/index.html}
+
+Now we can compile this to an HTML file:
+
+{[
+λ odoc html -I src -o . ./src/Player.odoc
```

Comment:
The `-o` is very important. I would explain that it's typically the root path of the generated documentation for a package.